lib/middleware.js

```diff
@@ -213,16 +213,31 @@ const getMiddlewareInstance = module.exports = function (opts) {
     function processRequest () {
       // This will hold the entire http response buffer
       let buf = Buffer.alloc(0)
+      let aborted = false
       let _write
 
       // We listen for the request to emit "end" or "finish" and then write to
-      // the cache since we have everything we need to do so
-      function onRequestFinished (err) {
-        if (err) {
-          log(
-            'request error. not parsing response for cache - %j',
-            err
-          )
+      // the cache since we have everything we need to do so. Also need to
+      // listen for the aborted event so we don't accidentally store
+      // empty/partial caches
+      req.on('aborted', () => {
+        // this is required for node v6, in node v8 we can use the
+        // "req" param passed to "onResponseFinished"
+        aborted = true
+      })
+      function onResponseFinished (err) {
+        if (err || aborted) {
```

Comment:
@evanshortiss Do you think it would be better to check for `res.finished` instead of `req.aborted` as that way if the connection was `close`d for any other reason it'd be caught... i.e. `req.finished` would be saying "only cache if the response has been correctly ended"?

The docs say "The method, response.end(), MUST be called on each response." and `req.finished` says "The response.finished property will be true if response.end() has been called." so think it might be safer?

Reply:
Additionally `res.finished` has been around since v0.0.2 so no need to worry about backwards compat.